For some reason, unciv seems to always try and build the android version.

Follow the AUR's fix and remove the android section completely.

--- build.gradle.kts.original	2023-02-18 12:59:42.000000000 +0000
+++ build.gradle.kts	2023-02-18 13:10:48.000000000 +0000
@@ -80,22 +80,6 @@
 
 }
 
-project(":android") {
-    apply(plugin = "com.android.application")
-    apply(plugin = "kotlin-android")
-
-    val natives by configurations.creating
-
-    dependencies {
-        "implementation"(project(":core"))
-        "implementation"("com.badlogicgames.gdx:gdx-backend-android:$gdxVersion")
-        "implementation"("org.jetbrains.kotlinx:kotlinx-coroutines-android:1.6.1")
-        natives("com.badlogicgames.gdx:gdx-platform:$gdxVersion:natives-armeabi-v7a")
-        natives("com.badlogicgames.gdx:gdx-platform:$gdxVersion:natives-arm64-v8a")
-        natives("com.badlogicgames.gdx:gdx-platform:$gdxVersion:natives-x86")
-        natives("com.badlogicgames.gdx:gdx-platform:$gdxVersion:natives-x86_64")
-    }
-}
 
 project(":ios") {
     apply(plugin = "kotlin")
